src/logic/map_objects/tribes/building.cc

```diff
@@ -844,20 +852,10 @@ void Building::set_soldier_control(SoldierControl* new_soldier_control) {
  * \note Warehouses always see their surroundings; this is handled separately.
  */
 void Building::set_seeing(bool see) {
-	if (see == seeing_) {
```

Comment:
Was this early return part of what caused the bug? If not, we might want to keep it to save us the update_vision call below.

Reply:
The bug*s* were on a deeper level. The new implementation requires this though or there will be vision bugs in some cornercases.